integrations/dd-sdk-android-glide/src/main/kotlin/com/datadog/android/glide/DatadogGlideModule.kt

```diff
@@ -88,13 +93,10 @@ constructor(
         val builder = OkHttpClient.Builder()
 
         builder.eventListenerFactory(DatadogEventListener.Factory(sdkInstanceName))
-        builder.addInterceptor(
-            DatadogInterceptor(
-                sdkInstanceName,
-                firstPartyHosts,
-                traceSampler = RateBasedSampler(sampleRate)
-            )
-        )
+        val interceptorBuilder = DatadogInterceptor.Builder(firstPartyHostsWithHeaderType)
+            .setTraceSampler(RateBasedSampler(sampleRate))
+        sdkInstanceName?.let { interceptorBuilder.setSdkInstanceName(it) }
```

Comment:
I would pull `sdkInstanceName` to the `DatadogInterceptor.Builder` constructor - SDK core association is a major part of the API, not something optional.

Reply:
in the same time we have it optional now, I would only do that if we do not allow it as optional anymore otherwise it doesn't make sense to have it in the constructor if it can be NULL